toothpick-runtime/src/main/java/toothpick/ToothPick.java

```diff
@@ -100,11 +107,27 @@ public static void inject(Object obj, Scope scope) {
     injector.inject(obj, scope);
   }
 
-  // Not synchronized, called by closeScope that is synchronized
+  /**
+   * Removes all nodes of {@code scope} using DFS. We don't lock here.
+   *
+   * @param scope the parent scope of which all children will recursively be removed
+   * from the map. We don't do anything else to the children nodes are they will be
+   * garbage collected soon. We just cut a whole sub-graph in the references graph of the JVM normally.
+   */
   private static void removeScopeAndChildrenFromMap(Scope scope) {
     MAP_KEY_TO_SCOPE.remove(scope.getName());
-    for (Scope childScope : scope.childrenScopes) {
+    for (Scope childScope : scope.childrenScopes.values()) {
```

Comment:
isn't there a way of going through the map values without the need of synchronized?